pkgs/development/python-modules/virtualenv/default.nix

```diff
@@ -1,25 +1,38 @@
 { buildPythonPackage
 , fetchPypi
 , lib
-, recursivePthLoader
+, appdirs
+, contextlib2
+, distlib
+, filelock
+, importlib-metadata
+, importlib-resources
+, pathlib2
+, setuptools_scm
+, six
 }:
 
 buildPythonPackage rec {
   pname = "virtualenv";
-  version = "16.7.9";
+  version = "20.0.21";
 
   src = fetchPypi {
     inherit pname version;
-    sha256 = "0d62c70883c0342d59c11d0ddac0d954d0431321a41ab20851facf2b222598f3";
-  };
-
-  # Doubt this is needed - FRidh 2017-07-07
-  pythonPath = [ recursivePthLoader ];
+    sha256 = "1kxnxxwa25ghlkpyrxa8pi49v87b7ps2gyla7d1h6kbz9sfn45m1";
 
-  patches = [ ./virtualenv-change-prefix.patch ];
+  };
 
-  # Tarball doesn't contain tests
-  doCheck = false;
+  propagatedBuildInputs = [
+    appdirs
+    contextlib2
+    distlib
+    filelock
+    importlib-metadata
+    importlib-resources
+    pathlib2
+    setuptools_scm
+    six
+  ];
```

Comment:
Some of these requirements could be added [conditionally on the Python version](https://github.com/pypa/virtualenv/blob/20.0.21/setup.cfg#L46-L49): contextlib2 is for Python < 3.3, pathlib2 is for Python < 3.4, importlib-resources is for Python < 3.7, importlib-metadata  is for Python < 3.8.

Reply:
Also setuptools_scm is used only at build time and needn’t be propagated.

Reply:
ok, will look into it after sleep :)

Reply:
And thanks @andersk for looking after it. I did totally miss this points.

Reply:
done in https://github.com/NixOS/nixpkgs/pull/88613/commits/86218dbcdc00ab17563514c1dd05971bbd9f5af8